app/src/main/java/com/example/pps_tudai/mvp/presenter/LoginPresenter.java

```diff
@@ -0,0 +1,46 @@
+package com.example.pps_tudai.mvp.presenter;
+
+import com.example.pps_tudai.mvp.model.LoginModel;
+import com.example.pps_tudai.mvp.view.LoginView;
+
+public class LoginPresenter {
+
+    private final LoginModel loginModel;
+    private final LoginView loginView;
+
+    public LoginPresenter(LoginModel loginModel, LoginView loginView) {
+        this.loginModel = loginModel;
+        this.loginView = loginView;
+    }
+
+    public void onLoginPressed() {
+        String email = loginView.getUserEmail();
+        String password = loginView.getPassword();
+
+        if (email.isEmpty() || password.isEmpty()) {
+            loginView.showDataEmptyScreen();
+        }
+        if(isvalid(email)) {
+            boolean isAuthenticated = loginModel.authenticateUser(email, password);
+            if (isAuthenticated) {
+                loginView.showWelcomeScreen();
+            } else {
+                loginView.showOnAuthenticationErrorMessage();
+            }
+        }
+        else{
+            loginView.showOnEmailErrorMessage();
+        }
+    }
+
+    private boolean isvalid(String email) {
+        String regex = "^[\\w-_\\.+]*[\\w-_\\.]\\@([\\w]+\\.)+[\\w]+[\\w]$";
```

Comment:
Put this regex in a constant pls. 
Maybe you can create a StringUtils class and put there all the constants, so you will be able to re-use from other places.

Reply:
done!